src/index/am/mod.rs

```diff
@@ -581,7 +588,7 @@ impl SearchFetcher for HeapFetcher {
             if !fetch_row_version(self.heap_relation, &mut ctid, self.snapshot, self.slot) {
                 return None;
             }
-            if !self.hack.is_null() && prererank_filtering() {
+            if !self.hack.is_null() && prefilter() {
```

Comment:
Why check it? Since checking and retrieving the vector are decoupled, this should not be done.